packages/commonwealth/client/scripts/views/pages/NotificationSettings/index.tsx

```diff
@@ -0,0 +1,180 @@
+import { PageLoading } from '../loading';
+import { SubscriptionEntry } from './SubscriptionEntry';
+import './index.scss';
+
+type NotificationSection = 'community-alerts' | 'subscriptions';
+
+const NotificationSettings = () => {
+  const navigate = useCommonNavigate();
+  const threadSubscriptions = useThreadSubscriptions();
+  const communityAlerts = useCommunityAlerts();
+  const enableKnockPushNotifications = useFlag('knockPushNotifications');
+  const registerClientRegistrationToken =
+    trpc.subscription.registerClientRegistrationToken.useMutation();
+
+  const communityAlertsIndex = createIndexForCommunityAlerts(
+    (communityAlerts.data as unknown as ReadonlyArray<
+      z.infer<typeof CommunityAlert>
+    >) || [],
+  );
+
+  const { bundledSubs } = useNotificationSettings();
+
+  const [threadsFilter, setThreadsFilter] = useState<readonly number[]>([]);
+  const [section, setSection] =
+    useState<NotificationSection>('community-alerts');
+
+  const handleUnsubscribe = useCallback(
+    (id: number) => {
+      setThreadsFilter([...threadsFilter, id]);
+    },
+    [threadsFilter],
+  );
+
+  const handlePushNotificationSubscription = useCallback(() => {
+    async function doAsync() {
+      const permission = await Notification.requestPermission();
+      if (permission === 'granted') {
+        console.log('Notification permission granted.');
+        const token = await getFirebaseMessagingToken();
+        await registerClientRegistrationToken.mutateAsync({
+          id: 'none',
+          token,
+        });
+      }
+    }
+
+    doAsync().catch(console.error);
+  }, [registerClientRegistrationToken]);
+
+  if (threadSubscriptions.isLoading) {
+    return <PageLoading />;
+  } else if (!app.isLoggedIn()) {
+    navigate('/', { replace: true });
+    return <PageLoading />;
+  }
+
+  return (
+    <CWPageLayout>
+      <div className="NotificationSettingsPage NotificationSettings">
+        <CWText type="h3" fontWeight="semiBold" className="page-header-text">
+          Notification settings
+        </CWText>
+
+        <CWText className="page-subheader-text">
+          Manage the emails and alerts you receive about your activity
+        </CWText>
+
+        {enableKnockPushNotifications && (
+          <div>
+            <CWText type="h5">Push Notifications</CWText>
+
+            <p>
+              <CWButton
+                label="Subscribe to Push Notifications"
+                onClick={handlePushNotificationSubscription}
+              />
+            </p>
+          </div>
+        )}
+
+        <CWTabsRow>
+          <CWTab
+            label="Community Alerts"
+            isSelected={section === 'community-alerts'}
+            onClick={() => setSection('community-alerts')}
+          />
+          <CWTab
+            label="Subscriptions"
+            isSelected={section === 'subscriptions'}
+            onClick={() => setSection('subscriptions')}
+          />
```

Comment:
Possible enhancement: we mostly add the active tab to URL, ex: `?tab=activeTabName` and use it to set the active tab. This also helps in page bookmarks etc...

Reply:
sounds good. I wouldn't saw its a blocker but would be great to get this in. If not lets create a ticket to address this later.

Reply:
OK created a ticket for it: 

https://github.com/hicommonwealth/commonwealth/issues/8191